File: sdk/android/story-sdk/src/main/kotlin/com/openstory/sdk/model/OpenStoryCtaPayload.kt
package com.openstory.sdk.model

data class OpenStoryCtaPayload(
    val placementKey: String,
    val storyGroupId: String,
    val storyGroupRevisionId: String,
    val storyId: String,
    val storyRevisionId: String,
    val label: String,
    val targetType: TargetType,
    val targetValue: String,
) {
    enum class TargetType {
        URL,
        DEEPLINK,
    }
}

File: sdk/ios/Sources/OpenStorySDK/Models/OpenStoryCtaPayload.swift
import Foundation

public struct OpenStoryCtaPayload: Sendable, Equatable {
    public enum TargetType: String, Sendable {
        case url
        case deeplink
    }

    public let placementKey: String
    public let storyGroupId: String
    public let storyGroupRevisionId: String
    public let storyId: String
    public let storyRevisionId: String
    public let label: String
    public let targetType: TargetType
    public let targetValue: String

    public init(
        placementKey: String,
        storyGroupId: String,
        storyGroupRevisionId: String,
        storyId: String,
        storyRevisionId: String,
        label: String,
        targetType: TargetType,
        targetValue: String
    ) {
        self.placementKey = placementKey
        self.storyGroupId = storyGroupId
        self.storyGroupRevisionId = storyGroupRevisionId
        self.storyId = storyId
        self.storyRevisionId = storyRevisionId
        self.label = label
        self.targetType = targetType
        self.targetValue = targetValue
    }
}
